protocol/daemons/slinky/client/price_fetcher.go

```diff
@@ -0,0 +1,143 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"cosmossdk.io/log"
+	"google.golang.org/grpc"
+
+	oracleclient "github.com/skip-mev/slinky/service/clients/oracle"
+	"github.com/skip-mev/slinky/service/servers/oracle/types"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+
+	"github.com/dydxprotocol/v4-chain/protocol/daemons/pricefeed/api"
+	daemontypes "github.com/dydxprotocol/v4-chain/protocol/daemons/types"
+)
+
+// PriceFetcher is responsible for pulling prices from the slinky sidecar and sending them to the pricefeed server.
+type PriceFetcher interface {
+	Start(ctx context.Context) error
+	Stop()
+	FetchPrices(ctx context.Context) error
+}
+
+// PriceFetcherImpl implements the PriceFetcher interface.
+type PriceFetcherImpl struct {
+	marketPairFetcher      MarketPairFetcher
```

Comment:
nit: this only needs Reader methods from the `MarketPairFetcher`? Why not j depend on an iface w/ that method specifically? [principle of least privilege](https://en.wikipedia.org/wiki/Principle_of_least_privilege)

Reply:
I didn't use an interface because they're in the same package anyway.